tensorflow/core/kernels/data/filter_by_component_dataset_op.cc

```diff
@@ -130,6 +130,10 @@ class FilterByLastComponentDatasetOp : public UnaryDatasetOpKernel {
             return Status::OK();
           }
 
+          if (out_tensors->back().NumElements() != 1) {
```

Comment:
Do we need to check the data type of the last component as well?

Reply:
yes, please check it is DT_BOOL